examples/matmul.mojo

```diff
@@ -32,7 +32,7 @@ alias type = DType.float32
 
 # simdwidth of = amount of `type` elements that fit into a single SIMD register
 # 2x multiplier will use multiple SIMD registers in parallel where possible
-alias nelts = simdwidthof[type]() * 2
+alias nelts = simdwidthof[type]() * 4
```

Comment:
This is faster on macos, but slower on macos/intel. Can it be changed to:

```mojo
# Get optimal SIMD_WIDTH at compile time.
# Helps with pipelining and running multiple SIMD operations in parallel.
alias nelts = get_simd_width()

fn get_simd_width() -> Int:
    @parameter
    if is_apple_silicon():
        return 4 * simdwidthof[FloatType]()
    else:
        return 2 * simdwidthof[FloatType]()
```

And moved below the tile alias' to keep the fn and alias close together